pkg/component/autoscaling/hvpa/crd.go

```diff
@@ -8,28 +8,82 @@ import (
 	"context"
 	_ "embed"
 
+	"golang.org/x/exp/maps"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/gardener/gardener/pkg/client/kubernetes"
 	"github.com/gardener/gardener/pkg/component"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
 )
 
-//go:embed templates/crd-autoscaling.k8s.io_hvpas.yaml
-var crd string
+var (
+	//go:embed templates/crd-autoscaling.k8s.io_hvpas.yaml
+	crdHvpas string
+
+	// resourceObjectKeyMap maps the ObjectKey of the CRD to its corresponding manifest
+	resourceObjectKeyMap map[client.ObjectKey]string
+)
+
+func init() {
+	resourceObjectKeyMap = make(map[client.ObjectKey]string)
+	resources := []string{crdHvpas}
+
+	for _, resource := range resources {
+		objKey, err := kubernetesutils.GetObjectKeyFromManifest(resource)
+		if err != nil {
+			panic(err)
+		}
```

Comment:
```suggestion
		utilruntime.Must(err)
```

Same for other occurrences.